### PR TITLE
Cypress. Update test case_4. 

### PR DESCRIPTION
**Pull request by dvkruchinin**
### Motivation and context
Test processing due to changes in the develop branch.
Added additional checks to the task deletion function.
A function for deleting users, tasks, and projects has been added to main.yml. This is due to the fact that Cypress did not catch the task deletion error in the test. The order of deleting tasks and users in the test has been changed to avoid dropping the test. The error was detected in https://github.com/openvinotoolkit/cvat/runs/4637596430?check_suite_focus=true.

### Checklist

- [x] I submit my changes into the `develop` branch
- [ ] I have added a description of my changes into [CHANGELOG](https://github.com/opencv/cvat/blob/develop/CHANGELOG.md) file
- [ ] I have updated the [documentation](
  https://github.com/opencv/cvat/blob/develop/README.md#documentation) accordingly
- [ ] I have added tests to cover my changes
- [ ] I have linked related issues ([read github docs](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword))
- [ ] I have increased versions of npm packages if it is necessary ([cvat-canvas](https://github.com/opencv/cvat/tree/develop/cvat-canvas#versioning),
[cvat-core](https://github.com/opencv/cvat/tree/develop/cvat-core#versioning), [cvat-data](https://github.com/opencv/cvat/tree/develop/cvat-data#versioning) and [cvat-ui](https://github.com/opencv/cvat/tree/develop/cvat-ui#versioning))

### License

- [x] I submit _my code changes_ under the same [MIT License](
  https://github.com/opencv/cvat/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.
- [x] I have updated the license header for each file (see an example below)

```python
# Copyright (C) 2021 Intel Corporation
#
# SPDX-License-Identifier: MIT
```

### CODE REVIEW

File: tests/cypress/integration/actions_users/registration_involved/case_4_assign_taks_job_users.js [hunk 1]
```diff
@@ -140,12 +150,11 @@ context('Multiple users. Assign task, job. Deactivating users.', () => {
         });
 
         // FIXME: the third user doesn't have permissions to open the task (only a job)
-        it.skip('Third user login. Tries to delete task. The task can be opened.', () => {
+        it('A third user can open a job by a direct link.', () => {
```
Comment (bsekachev) [inline tests/cypress/integration/actions_users/registration_involved/case_4_assign_taks_job_users.js:153]:
```suggestion
        it('The third user can open a job by a direct link.', () => {
```

File: tests/cypress/integration/actions_users/registration_involved/case_4_assign_taks_job_users.js [hunk 2]
```diff
@@ -77,7 +80,7 @@ context('Multiple users. Assign task, job. Deactivating users.', () => {
                 secondUser.password,
             );
             cy.createAnnotationTask(
-                taskName,
+                `${taskName} second`,
```
Comment (bsekachev) [inline tests/cypress/integration/actions_users/registration_involved/case_4_assign_taks_job_users.js:83]: Maybe create another identificator to store the second task name?

File: tests/cypress/integration/actions_users/registration_involved/case_4_assign_taks_job_users.js [hunk 3]
```diff
@@ -109,6 +112,13 @@ context('Multiple users. Assign task, job. Deactivating users.', () => {
...
             cy.login();
             cy.createAnnotationTask(taskName, labelName, attrName, textDefaultValue, archiveName);
+            cy.goToTaskList();
+            cy.openTaskJob(taskName);
+            // Getting the task and job id
+            cy.url().then((url) => {
+                jobID = Number(url.split('/').slice(-1)[0]);
+                taskID = Number(url.split('/').slice(-3)[0]);
+            });
             cy.logout();
```
Comment (bsekachev) [inline tests/cypress/integration/actions_users/registration_involved/case_4_assign_taks_job_users.js:122]: Should we really logout before the previous it()? We login again there

Comment (bsekachev) [reply tests/cypress/integration/actions_users/registration_involved/case_4_assign_taks_job_users.js:122]: Maybe unite these cases?

File: tests/cypress/integration/actions_users/registration_involved/case_4_assign_taks_job_users.js (hunk 1 above)
Comment (dvkruchinin) [reply tests/cypress/integration/actions_users/registration_involved/case_4_assign_taks_job_users.js]: Applied locally. Thanks.

File: tests/cypress/integration/actions_users/registration_involved/case_4_assign_taks_job_users.js (hunk 2 above)
Comment (dvkruchinin) [reply tests/cypress/integration/actions_users/registration_involved/case_4_assign_taks_job_users.js]: Done.

File: tests/cypress/integration/actions_users/registration_involved/case_4_assign_taks_job_users.js (hunk 3 above)
Comment (dvkruchinin) [reply tests/cypress/integration/actions_users/registration_involved/case_4_assign_taks_job_users.js]: Sure. Thanks.

**Review by bsekachev — APPROVED** [review]

**Pull request merged**